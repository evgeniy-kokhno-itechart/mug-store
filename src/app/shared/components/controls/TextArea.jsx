import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../markup/FormGroup';
import ControlBase from './ControlBase';

class TextArea extends ControlBase {
  render() {
    const {
      name, label, value, error,
    } = this.props;

    return (
      <FormGroup name={name} label={label} error={error}>
        <textarea name={name} id={name} onChange={this.handleChange} className="form-control" value={value} />
      </FormGroup>
    );
  }
}

TextArea.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  error: PropTypes.string,
};

TextArea.defaultProps = {
  value: '',
  error: '',
};

export default TextArea;
