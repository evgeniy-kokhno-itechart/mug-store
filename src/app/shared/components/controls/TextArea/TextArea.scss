.formgroup {
    &__text-area {
        font: inherit;
        width: 100%;
        padding: 0.4rem 0.5rem;
    }
}