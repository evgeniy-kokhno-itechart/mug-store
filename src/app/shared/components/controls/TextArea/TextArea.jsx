import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../../markup/FormGroup/FormGroup';
import ControlBase from '../ControlBase';
import './TextArea.scss';

class TextArea extends ControlBase {
  render() {
    const {
      name, label, value, error,
    } = this.props;

    return (
      <FormGroup name={name} label={label} error={error}>
        <textarea name={name} id={name} onChange={this.handleChange} className="formgroup__text-area focusable" value={value} />
      </FormGroup>
    );
  }
}

TextArea.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  error: PropTypes.string,
};

TextArea.defaultProps = {
  value: '',
  error: '',
};

export default TextArea;
