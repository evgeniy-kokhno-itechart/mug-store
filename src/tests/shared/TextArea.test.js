/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TextArea from '../../shared/controls/TextArea';

configure({ adapter: new Adapter() });

describe('<TextArea />', () => {
  let TextAreaWrapper;

  beforeEach(() => {
    const onValueChange = jest.fn();
    TextAreaWrapper = shallow(<TextArea name="testname" label="Test label" onValueChange={onValueChange} />);
  });

  it('check TextArea renders properly with default props', () => {
    expect(TextAreaWrapper.debug()).toMatchSnapshot();
  });

  it('check TextArea renders properly with error', () => {
    TextAreaWrapper.setProps({ error: 'Test error' });
    expect(TextAreaWrapper.debug()).toMatchSnapshot();
  });

  it('check TextArea renders properly with value provided', () => {
    TextAreaWrapper.setProps({ value: 'Test value' });
    expect(TextAreaWrapper.debug()).toMatchSnapshot();
  });
});
