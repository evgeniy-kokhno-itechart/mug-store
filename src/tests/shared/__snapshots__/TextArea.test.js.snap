// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> check TextArea renders properly with default props 1`] = `
"<div className=\\"form-group\\">
  <label htmlFor=\\"testname\\">
    Test label
  </label>
  <textarea name=\\"testname\\" id=\\"testname\\" onChange={[Function: onChange]} className=\\"form-control\\" value=\\"\\" />
</div>"
`;

exports[`<TextArea /> check TextArea renders properly with error 1`] = `
"<div className=\\"form-group\\">
  <label htmlFor=\\"testname\\">
    Test label
  </label>
  <textarea name=\\"testname\\" id=\\"testname\\" onChange={[Function: onChange]} className=\\"form-control\\" value=\\"\\" />
  <div className=\\"alert alert-danger\\">
    Test error
  </div>
</div>"
`;

exports[`<TextArea /> check TextArea renders properly with value provided 1`] = `
"<div className=\\"form-group\\">
  <label htmlFor=\\"testname\\">
    Test label
  </label>
  <textarea name=\\"testname\\" id=\\"testname\\" onChange={[Function: onChange]} className=\\"form-control\\" value=\\"Test value\\" />
</div>"
`;
